code/modules/integrated_electronics/subtypes/output.dm

```diff
@@ -34,6 +34,34 @@
 	else
 		stuff_to_display = replacetext("[I.data]", eol , "<br>")
 
+/obj/item/integrated_circuit/output/screen/large
+	name = "large screen"
+	desc = "Takes any data type as an input and displays it to anybody near the device when pulsed. \
+	It can also be examined to see the last thing it displayed."
+	icon_state = "screen_medium"
+	power_draw_per_use = 20
+
+/obj/item/integrated_circuit/output/screen/large/do_work()
+	..()
+
+	var/in_hands = FALSE//this whole block just returns if the assembly is neither in a mobs hands or on the ground
+	if(isliving(assembly.loc))
+		in_hands = TRUE
+		var/mob/living/H = assembly.loc
+		if(H.get_active_held_item() != assembly && H.get_inactive_held_item() != assembly)
```

Comment:
can you explain the point of `in_hands` when this line exists?

Reply:
sure! if you look further down at line 53, i need to check a second time whether or not the assembly is in hands. if i don't, then at line 53 it will return the whole proc even if it passes this check here at line 51. in_hands is set to TRUE here so that if this part succeeds, then the check at line 53 will also succeed. you'll notice it's set to TRUE before the check for whether or not it actually is in hands or not happens: this is because if the check fails, then the proc returns anyways despite that var being TRUE.
tl;dr code logic reasons

Reply:
basically to make it simpler, if `in_hands` didn't exist then even if the object actually IS in somebody's hands, the code would still return at line 53. so we need it to make that not happen

Reply:
it's probably just because you never actually use it, so it looks weird to devs

stylistically you could do
```
var/in_hands = (H.get_active_held_item() = assembly || H.get_inactive_held_item() = assembly)
if(!in_hands)
   return
if(!isturf(assembly.loc) && !in_hands)
   return
```
but then it's obvious that you don't need the first check?

I'm confused